### PR TITLE
Bucket Policy Only Samples

### PR DESCRIPTION
**Pull request by frankyn**
These are samples to support Bucket Policy Only Samples.

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

I can't assign you @chestercun, PTAL. Thank you.

### CODE REVIEW

File: samples/buckets.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ async function createBucket(bucketName) {
 async function listBuckets() {
   // [START storage_list_buckets]
   // Imports the Google Cloud client library
-  const {Storage} = require('@google-cloud/storage');
+  const { Storage } = require('@google-cloud/storage');
```
Comment (JustinBeckwith) [inline samples/buckets.js]: Looks like you need to run `npm run fix` :)

File: samples/system-test/buckets.test.js [hunk 2]
```diff
@@ -86,6 +86,68 @@ it(`should set a bucket's default KMS key`, async () => {
...
+  );
+
+  assert.strictEqual(
+    (results.stdout + results.stderr).includes(
+      `Bucket Policy Only is enabled for ${bucketName}.`
+    ),
+    true
+  );
+  const [metadata] = await bucket.getMetadata();
+  assert.strictEqual(metadata.iamConfiguration.bucketPolicyOnly.enabled, true);
```
Comment (kinwa91) [inline samples/system-test/buckets.test.js]: (style)
`assert.ok(metadata.iamConfiguration.bucketPolicyOnly.enabled);`

Comment (frankyn) [reply samples/system-test/buckets.test.js]: Nice! I'll fix that, thanks @kinwa91!

**Comment by kinwa91** [comment]
@frankyn looks like we've never return `Date` object for `datetime` strings in nodejs-storage.

@stephenplusplus have we ever discussed whether to convert `datetime` strings to `Date` object in handwritten libraries before?

**Comment by stephenplusplus** [comment]
Sorry I missed this question. I think our rule has been to simplify a handwritten response format as best we can, and leave the API's raw response format as it came.

Considering a made-up method:

```js
file.getUsefulInfo((err, info, apiResponse) => {
  info.createdAt === Date('...')
  info.size === Number(8800)

  apiResponse.createdAt === 'long-string-of-numbers'
  apiResponse.size === '8800'
})
```

So, if a method exists and we can be more helpful by returning a handwritten response, we should be. For the `apiResponse`, I think we should leave it be, as opposed to iterating over it with best effort type conversion, as we might run into some unintentional side effects.

**Comment by frankyn** [comment]
Thanks @stephenplusplus and @kinwa91! Let's keep as-is. I'll resolve conflicts and merge **after** the workaround in this PR is no longer needed, e.g. Tomorrow EOD (Wednesday).

**Comment by frankyn** [comment]
@kinwa91 are you okay with I merge?

**Pull request merged**